src/CommunityToolkit/Xamarin.CommunityToolkit/Converters/BaseConverterOneWay.shared.cs

```diff
@@ -20,14 +25,20 @@ public abstract class BaseConverterOneWay<TFrom, TTo> : ValueConverterExtension,
 		/// <param name="parameter">Additional parameter for the converter to handle. This is not implemented.</param>
 		/// <param name="culture">The culture to use in the converter. This is not implemented.</param>
 		/// <returns>An object of type <see cref="TTo"/>.</returns>
-		public object? Convert(object value, Type targetType, object parameter, CultureInfo culture)
+		public object? Convert(object? value, Type targetType, object parameter, CultureInfo culture)
 		{
-			if (value is not TFrom valueFrom)
-				throw new ArgumentException($"value needs to be of type {typeof(TFrom)}");
-
 			if (targetType != typeof(TTo) && !(typeof(TFrom) != typeof(string)))
 				throw new ArgumentException($"targetType needs to be typeof {typeof(TTo)}");
 
+			if (AllowsNullOrDefault is false && value is null)
+				throw new ArgumentNullException($"value needs to be of type {typeof(TFrom)} but is null. Consider setting {nameof(AllowsNullOrDefault)} to true.");
+
+			if (AllowsNullOrDefault is true && value is null)
```

Comment:
AllowsNullOrDefault is a Boolean, so we don't need to check it
```suggestion
			if (AllowsNullOrDefault && value is null)
```

Reply:
Perhaps something like this might be easier to read?

```csharp
if (value is null)
{
    if (AllowsNullOrDefault)
        throw new ArgumentNull...

    return Convert(default);
}
```

Reply:
love it /\

Reply:
I've changed it but swapped them out (inverted).